workflows/trust4/trust4.wdl

```diff
@@ -0,0 +1,298 @@
+            String genome_file 
+            Int num_cpu
+            String memory
+            Int disk_space
+            String se_fastq_pattern
+            String? pe_read1_fastq_pattern
+            String? pe_read2_fastq_pattern
+            String? read1_range
+            String? read2_range
+            String barcode_fastq_pattern
+            String barcode_range
+            File? barcode_whitelist
+            String umi_fastq_pattern
+            String umi_range
+            String? umi_bam_field
+            File? input_bam
+            String? bam_barcode_field
+            String? bam_abnormal_unmap_flag
+            Boolean? skipMateExtension
+            Int? mateIdSuffixLen
+            Boolean? noExtraction
+            String? repseq
+            Boolean? outputReadAssignment
+            String docker_registry
+            String zones
+            Int preemptible
+            Int awsMaxRetries
+            String backend
+    }
+
+    command {
+        set -e
+        export TMPDIR=/tmp
+        monitor_script.sh > monitoring.log &
+        mkdir -p genome_dir
+        tar xf ~{genome_file} -C genome_dir --strip-components 1   
+
+        python <<CODE
+        import re
+        import os
+        from fnmatch import fnmatch
+        from subprocess import check_call, CalledProcessError, DEVNULL, STDOUT
+        import sys
+
+        Path = os.path.join
+
+        if '~{input_fastqs_directories}' and '~{input_bam}':
+            sys.exit("Please provide either Fastq OR BAM as input (not both)")
+
+        if '~{input_fastqs_directories}':
+            if '~{pe_read1_fastq_pattern}' and '~{pe_read2_fastq_pattern}':
+                se_fastq_pattern = ''
+            else:         
+                se_fastq_pattern = '~{se_fastq_pattern}'
+
+        if '~{input_fastqs_directories}':
+            fastq_dirs = []
+            for i, directory in enumerate('~{input_fastqs_directories}'.split(',')):
+                directory = re.sub('/+$', '', directory) # remove trailing slashes
+                target = "~{sample_id}_" + str(i)
+                try:
+                    call_args = ['strato', 'exists', '--backend', '~{backend}', directory + '/~{sample_id}/']
+                    print(' '.join(call_args))
+                    check_call(call_args, stdout=DEVNULL, stderr=STDOUT)
+                    call_args = ['strato', 'sync', '--backend', '~{backend}', '-m', directory + '/~{sample_id}', target]
+                    print(' '.join(call_args))
+                    check_call(call_args)     
+                except CalledProcessError:
+                    if not os.path.exists(target):
+                        os.mkdir(target)
+                    call_args = ['strato', 'cp', '--backend', '~{backend}', '-m', directory + '/~{sample_id}*' , target + '/']
+                    print(' '.join(call_args))
+                    check_call(call_args)
+
+                fastq_dirs.append(target)
+
+        if '~{input_bam}':
```

Comment:
~{input_bam} itself is a BAM file, there is no need to copy anything from bucket.